packages/nimble-components/src/utilities/unit-format/decimal-unit-format.ts

```diff
@@ -0,0 +1,76 @@
+import { UnitFormat } from './base/unit-format';
+import type { ScaledUnitFormat } from './unit-scale/base/scaled-unit-format';
+import type { UnitScale } from './unit-scale/base/unit-scale';
+import { passthroughUnitScale } from './unit-scale/passthrough-unit-scale';
+
+/**
+ * Format for decimal numbers with units.
+ */
+export class DecimalUnitFormat extends UnitFormat {
+    /** Resolved UnitScale */
+    public readonly unitScale: UnitScale;
+
+    /** Resolved minimumFractionDigits */
+    public readonly minimumFractionDigits: number;
+
+    /** Resolved maximumFractionDigits */
+    public readonly maximumFractionDigits: number;
+
+    private readonly scaledUnitFormatters = new Map<number, ScaledUnitFormat>();
+    private readonly tenPowDecimalDigits: number;
+
+    public constructor(
+        locale: string,
+        {
+            minimumFractionDigits = 0,
+            maximumFractionDigits = Math.max(3, minimumFractionDigits),
+            unitScale = passthroughUnitScale
+        }: {
+            minimumFractionDigits?: number,
+            maximumFractionDigits?: number,
+            unitScale?: UnitScale
+        } = {
+            minimumFractionDigits: 0,
+            maximumFractionDigits: 3,
+            unitScale: passthroughUnitScale
+        }
```

Comment:
Yep, it's the destructuring assignment language feature. Reduces a lot of boilerplate for configuration objects as you described. MDN is a bit terse but this gives some discussion https://javascript.info/destructuring-assignment#smart-function-parameters
The named types instead of the inline ones also make it a little clearer